promutils/labeled/counter.go

```diff
@@ -59,8 +73,10 @@ func NewCounter(name, description string, scope promutils.Scope, opts ...MetricO
 		if _, emitUnlabeledMetric := opt.(EmitUnlabeledMetricOption); emitUnlabeledMetric {
 			c.Counter = scope.MustNewCounter(GetUnlabeledMetricName(name), description)
 		} else if additionalLabels, casted := opt.(AdditionalLabelsOption); casted {
-			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, additionalLabels.Labels...)...)
-			c.additionalLabels = contextutils.MetricKeysFromStrings(additionalLabels.Labels)
+			labels := GetUniqueLabels(metricStringKeys, additionalLabels.Labels)
+			// Here we only append the labels that don't exist in metricStringKeys
+			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, labels...)...)
```

Comment:
This `append` here may modify the same underly array backing up `metricStringKeys` and cause corruption for a counter that was created previously.

A repro:

```go
package labeled

import (
	"context"
	"testing"

	"github.com/flyteorg/flytestdlib/contextutils"
	"github.com/flyteorg/flytestdlib/promutils"
	"github.com/stretchr/testify/assert"
)

func TestLabeledCounter(t *testing.T) {
	UnsetMetricKeys()
	assert.NotPanics(t, func() {
		SetMetricKeys(contextutils.ProjectKey, contextutils.DomainKey, contextutils.WorkflowIDKey)
	})

	scope := promutils.NewTestScope()
	c1 := NewCounter(
		"handle_count1",
		"c1",
		/*scope=*/ scope,
		/*opts=*/ AdditionalLabelsOption{Labels: []string{"foo"}},
	)
	c2 := NewCounter(
		"handle_count2",
		"c2",
		/*scope=*/ scope,
		/*opts=*/ AdditionalLabelsOption{Labels: []string{"tasktype"}}, // this corrupts c1
	)

	ctx := context.TODO()
	ctx = contextutils.WithProjectDomain(ctx, "project", "domain")
	ctx = contextutils.WithWorkflowID(ctx, "workflow")
	ctx = contextutils.WithTaskID(ctx, "task")
	ctx = contextutils.WithTaskType(ctx, "type")
	ctx = context.WithValue(ctx, "foo", "foo")

	c1.Inc(ctx) // <--- panic here
	c2.Inc(ctx)
}
```

cc @hamersaw

Reply:
It was not introduced by this PR though, the last change of this was ages ago in https://github.com/pingsutw/flytestdlib/commit/c811acb9176b93f55fce4f8a6f3d0339cee0f9de. :)